packages/ics721/src/testing/contract.rs

```diff
@@ -549,3 +566,95 @@ fn test_instantiate() {
+        cw721_base_code_id: Some(1),
+        cw721_admin: Some("some_other_admin".to_string()),
+    };
+
+    // before migrate, populate legacy
+    let class_id_to_nft_contract: Map<ClassId, Addr> = Map::new("e");
+    class_id_to_nft_contract
+        .save(
+            deps.as_mut().storage,
+            ClassId::new(CLASS_ID_1),
+            &Addr::unchecked(NFT_CONTRACT_1),
+        )
+        .unwrap();
+    class_id_to_nft_contract
+        .save(
+            deps.as_mut().storage,
+            ClassId::new(CLASS_ID_2),
+            &Addr::unchecked(NFT_CONTRACT_2),
+        )
+        .unwrap();
+    let nft_contract_to_class_id: Map<Addr, ClassId> = Map::new("f");
+    nft_contract_to_class_id
+        .save(
+            deps.as_mut().storage,
+            Addr::unchecked(NFT_CONTRACT_1),
+            &ClassId::new(CLASS_ID_1),
+        )
+        .unwrap();
+    nft_contract_to_class_id
+        .save(
+            deps.as_mut().storage,
+            Addr::unchecked(NFT_CONTRACT_2),
+            &ClassId::new(CLASS_ID_2),
+        )
+        .unwrap();
+
+    // migrate
+    Ics721Contract {}
+        .migrate(deps.as_mut(), env.clone(), msg)
+        .unwrap();
+
+    assert_eq!(
+        PO.pauser.load(&deps.storage).unwrap(),
+        Some(Addr::unchecked("some_other_pauser"))
+    );
+    assert_eq!(
+        OUTGOING_PROXY.load(&deps.storage).unwrap(),
+        Some(Addr::unchecked("outgoing"))
+    );
+    assert_eq!(
+        INCOMING_PROXY.load(&deps.storage).unwrap(),
+        Some(Addr::unchecked("incoming"))
+    );
+    assert_eq!(CW721_CODE_ID.load(&deps.storage).unwrap(), 1);
+    assert_eq!(
+        ADMIN_USED_FOR_CW721.load(&deps.storage).unwrap(),
+        Some(Addr::unchecked("some_other_admin"))
+    );
+    let nft_contract_and_class_id_list = query_nft_contracts(deps.as_ref(), None, None).unwrap();
+    assert_eq!(nft_contract_and_class_id_list.len(), 2);
+    assert_eq!(nft_contract_and_class_id_list[0].0, CLASS_ID_1);
+    assert_eq!(nft_contract_and_class_id_list[0].1, NFT_CONTRACT_1);
+    assert_eq!(nft_contract_and_class_id_list[1].0, CLASS_ID_2);
+    assert_eq!(nft_contract_and_class_id_list[1].1, NFT_CONTRACT_2);
+    // test query and indexers for class id and addr are working
+    let nft_contract_1 =
+        query_nft_contract_for_class_id(&deps.storage, CLASS_ID_1.to_string()).unwrap();
+    assert_eq!(nft_contract_1, Some(Addr::unchecked(NFT_CONTRACT_1)));
+    let nft_contract_2 =
+        query_nft_contract_for_class_id(&deps.storage, CLASS_ID_2.to_string()).unwrap();
+    assert_eq!(nft_contract_2, Some(Addr::unchecked(NFT_CONTRACT_2)));
+    let class_id_1 =
+        query_class_id_for_nft_contract(deps.as_ref(), NFT_CONTRACT_1.to_string()).unwrap();
+    assert_eq!(class_id_1, Some(ClassId::new(CLASS_ID_1)));
+    let class_id_2 =
+        query_class_id_for_nft_contract(deps.as_ref(), NFT_CONTRACT_2.to_string()).unwrap();
+    assert_eq!(class_id_2, Some(ClassId::new(CLASS_ID_2)));
+}
```

Comment:
test migration by populating new `CLASS_ID_AND_NFT_CONTRACT_INFO` with legacy data from `NFT_CONTRACT_TO_CLASS_ID`. Note: `CLASS_ID_TO_NFT_CONTRACT` is not used, since it is redundant to `NFT_CONTRACT_TO_CLASS_ID`